backend/tests/BDD/step_defs/test_prompts.py

```diff
@@ -0,0 +1,85 @@
+from pytest_bdd import given, when, then, parsers, scenarios
+import pytest
+import logging
+from tests.BDD.test_utilities import (
+    send_prompt,
+    app_healthcheck,
+    correctness_evaluator,
+    healthy_response,
+    check_response_confidence,
+)
+from decimal import Decimal
+import decimal
+
+logger = logging.getLogger(__name__)
+
+scenarios("../features/Correctness/Accuracy_Factual_Correctness.feature")
+
+
+@pytest.fixture
+def context():
+    return {}
+
+
+@given(parsers.parse("a prompt to InferESG"))
+def prepare_prompt(context):
+    healthcheck_response = app_healthcheck()
+    assert healthcheck_response.status_code == 200
+    assert healthcheck_response.json() == healthy_response
+    context["health_check_passed"] = True
+
+
+@when(parsers.parse("I get the response"))
+def get_response(context):
+    assert context.get("health_check_passed", False)
+
+
+@then(parsers.parse("the response to this '{prompt}' should match the '{expected_response}'"))
+def check_response_includes_expected_response(context, prompt, expected_response):
+    response = send_prompt(prompt)
+    actual_response = response.json()
+
+    try:
+        expected_value = Decimal(str(expected_response).strip())
+        actual_value = Decimal(str(actual_response).strip())
+
+        tolerance = Decimal("0.01")
+        is_equal = abs(expected_value - actual_value) <= tolerance
+
```

Comment:
We may not actually need this decimal stuff. I originally thought we had an issue with types before finding out it was the closed loop error stuff. Don't think it would hurt to keep it in though!

Reply:
Yes, I don't think it's currently used, as it would only apply when just a numeric value is returned.